### PR TITLE
feat(pcc.user): add possibility to edit an active directory

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pcc-federation` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-10028 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added the possibility to edit an active directory

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/federation-edit.constant.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  username: 'vcenterbind@example.com',
+  description: 'Example Active Directory',
+  sslThumbprint: 'BB:46:CA:6B:FC:92:4E:96:B4:BB:6E:44:7E:8F:AD:4C:C9:32:AB:AB',
+  password: '************',
+};
+
+const FIELD = 'Field';
+
+export const FIELD_NAME = {
+  username: `userName${FIELD}`,
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/federation-edit.constant.js:11]: Pascal name for variable: userName

Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/federation-edit.constant.js]: It's applied

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import angular from 'angular';
+import 'angular-translate';
+
+import component from './federation-edit.component';
+import taskPollProgress from '../../../taskPollProgress';
+
+const moduleName = 'ovhManagerDedicatedCloudUsersFederationEdit';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js:7]:
```suggestion
const moduleName = 'ovhManagerDedicatedCloudUsersActiveDirectoriesEdit';
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+angular
+  .module(moduleName, [taskPollProgress])
+  .component('dedicatedCloudUsersFederationEdit', component)
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js:11]:
```suggestion
  .component('dedicatedCloudUsersActiveDirectoriesEdit', component)
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.module.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import angular from 'angular';
+
+import routing from './federation-edit.routes';
+import federationEditComponent from '../../../../components/dedicated-cloud/users/activeDirectories/edit';
+
+const moduleName = 'dedicatedCloudUsersFederationEditModule';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.module.js:6]:
```suggestion
const moduleName = 'dedicatedCloudUsersActiveDirectoriesEditFederationEdit';
```

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.routes.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.dedicatedCloud.details.users.federation-edit', {
+    url: '/federation/edit',
+    params: {
+      activeDirectory: null,
+    },
+    views: {
+      modal: {
+        component: 'dedicatedCloudUsersFederationEdit',
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.routes.js:9]: to update regarding my previous comment.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/index.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import angular from 'angular';
+import '@uirouter/angularjs';
+import 'oclazyload';
+
+const moduleName = 'dedicatedCloudUsersFederationEditLazyloading';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/index.js:5]:
```suggestion
const moduleName = 'dedicatedCloudUsersActiveDirectoriesEditLazyloading';
```

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/edit/federation-edit.module.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import angular from 'angular';
+
+import routing from './federation-edit.routes';
+import federationEditComponent from '../../../../components/dedicated-cloud/users/activeDirectories/edit';
+
+const moduleName = 'managedBaremetalUsersFederationEditModule';
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/edit/federation-edit.module.js:6]:
```suggestion
const moduleName = 'managedBaremetalUsersActiveDirectoriesEditFederationEdit';
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js (hunk 3 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/activeDirectories/edit/index.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.module.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.module.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/index.js (hunk 6 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/index.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/edit/federation-edit.module.js (hunk 7 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/managedBaremetal/users/activeDirectories/edit/federation-edit.module.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.routes.js (hunk 5 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicatedCloud/users/activeDirectories/edit/federation-edit.routes.js]: It's applied.

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**